listener/reconciler/clusteraccess/reconciler.go

```diff
@@ -90,13 +87,15 @@ func CheckClusterAccessCRDStatus(ctx context.Context, k8sClient client.Client, l
 
 // ClusterAccessReconciler handles reconciliation for ClusterAccess resources
 type ClusterAccessReconciler struct {
-	lifecycleManager *lifecycle.LifecycleManager
-	opts             reconciler.ReconcilerOpts
+	client.Client
+	Scheme           *runtime.Scheme
```

Comment:
Why is the `Scheme` an exported field? This alerts my radar for unwanted side-effects :D